tools/licenses/lib/main.dart

```diff
@@ -2357,8 +2359,21 @@ Future<Null> main(List<String> arguments) async {
         if (signature != null)
           sink.writeln('Signature: $signature\n');
 
+        RepositoryDirectory componentRoot;
+        if (isFirstComponent) {
+          // For the first component, we can use the results of the initial
+          // repository crawl.
+          isFirstComponent = false;
+          componentRoot = component;
+        } else {
+          // For other components, we need a clean repository that does not
+          // contain any state left over from previous components.
+          clearLicenseRegistry();
+          componentRoot = (new RepositoryRoot(rootDirectory)).subdirectories.firstWhere(
```

Comment:
style nit: i believe there is no need for the parens around the constructor expression
style nit: since you have a newline after the open paren, there should be a matching newline before the closing paren.